packages/firestore/src/lite/transaction.ts

```diff
@@ -128,12 +134,12 @@ export class Transaction {
    */
   set<T>(
     documentRef: DocumentReference<T>,
-    data: Partial<T>,
+    data: PartialWithFieldValue<T>,
     options: SetOptions
   ): this;
   set<T>(
     documentRef: DocumentReference<T>,
-    value: T,
+    value: PartialWithFieldValue<T>,
```

Comment:
While we're at this, can we create a new overload for `value: PartialWithFieldValue<T>, options?: SetOptions & {merge: true}` but keep `value` non-partial otherwise (e.g. merge not specified)?

Reply:
I talked to Brian about this offline and just wanted to make sure this is captured here. While correct, this change would break users that use `merge: boolean` as the input type and make it cumbersome to define the merge behavior programatically. For this reason, I think we should hold off on making this change.